src/utils/error_handling.cpp

```diff
@@ -126,3 +125,55 @@ void Check_Configuration(parameters const &P)
 
 #endif  // MHD
 }
+
+// NOLINTNEXTLINE(cert-dcl50-cpp)
+[[noreturn]] void Abort_With_Err_(const char* func_name, const char* file_name, int line_num, const char* msg, ...)
+{
+  // considerations when using MPI:
+  //  - all processes must execute this function to catch errors that happen on
+  //    just one process
+  //  - to handle cases where all processes encounter the same error, we
+  //    pre-buffer the error message (so that the output remains legible)
+
+  // since we are aborting, it's OK that this isn't the most optimized
+
+  // prepare some info for the error message header
+  const char* santized_func_name = (func_name == nullptr) ? "{unspecified}" : func_name;
+
+  std::string proc_info =
+#ifdef MPI_CHOLLA
+      std::to_string(procID) + " / " + std::to_string(nproc) + " (using MPI)";
+#else
+      "0 / 1 (NOT using MPI)"
+#endif
+
+  // prepare the formatted message
+  std::vector<char> msg_buf;
+  if (msg == nullptr) {
+    msg_buf = std::vector<char>(80);
```

Comment:
I say this because I had to rewrite a bunch of code awhile ago because of a hard coded `char *` limit that was too low and it was super annoying; that limit was much higher than 80.

Reply:
Yeah, I was being lazy here. It's now not an issue anymore (now that we are using ``std::string``)

As an aside - we probably wouldn't have run into any issues here since this was a buffer for a fallback hardcoded message (but it still shouldn't have been hardcoded - just in case anybody ever altered the hardcoded message).